marklogic-data-hub/src/test/java/com/marklogic/hub/flow/impl/StemmingTest.java

```diff
@@ -0,0 +1,102 @@
+
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class StemmingTest extends AbstractHubCoreTest {
+
+    Database db;
+    @Autowired
+    FlowRunnerImpl flowRunner;
+
+    @BeforeEach
+    public void setupEach() {
+        HubConfig hubConfig = runAsAdmin();
+        db = new Database(new API(hubConfig.getManageClient()), hubConfig.getDbName(DatabaseKind.STAGING));
+        db.setStemmedSearches("advanced");
+        db.save();
+        installProjectInFolder("flow-runner-test");
+    }
+    /*
+     https://bugtrack.marklogic.com/54003
+     When stemmed search is set to "advanced", fetching legacy mappings (and/or step definition) fails and hence running
+     testFlow would fail. After that if db is reindexed, fetching of artifacts succeeds. In order to apply the fix for
+     bug 54003 in server, "Reindex JSON language property” trace event has to be added and the test should be run on
+     nightly servers. With the fix applied, the flow will fail after reindexing as well as queries requires 'lang=zxx'
+     to be included so as to fetch artifacts.
+     */
+    @Test
+    void runFlowWithStemmingSearch() {
+        Versions.MarkLogicVersion mlVersion = versions.getMLVersion();
+        Assumptions.assumeTrue(mlVersion.getMajor() >=10);
+
+        try{
+            RunFlowResponse resp = runFlow();
+            Assertions.assertTrue(JobStatus.STOP_ON_ERROR.toString().equalsIgnoreCase(resp.getJobStatus()));
+
+            HubConfig hubConfig = runAsAdmin();
+            DatabaseManager dbManager = new DatabaseManager(hubConfig.getManageClient());
+            dbManager.reindexDatabase(hubConfig.getDbName(DatabaseKind.STAGING));
+
+            waitForReIndex(runAsAdmin());
+            resp = runFlow();
+            Assertions.assertTrue(JobStatus.FINISHED.toString().equalsIgnoreCase(resp.getJobStatus()));
+
+        }
+        finally {
+            db.setStemmedSearches("off");
+            db.save();
+        }
+    }
+
+    private RunFlowResponse runFlow(){
+        runAsDataHubOperator();
+        RunFlowResponse resp = null;
+        try{
+            resp = flowRunner.runFlow("testFlow", null, null, null, null);
+        }
+        catch (Exception e){
+            logger.error("Flow run failed;cause: "+ e.getMessage());
+        }
+
+        flowRunner.awaitCompletion();
+        return resp;
+    }
+
+    private void waitForReIndex(HubConfig hubConfig){
+        String query = "(\n" +
+            "  for $forest-id in xdmp:database-forests(xdmp:database('data-hub-STAGING'))\n" +
+            "  return xdmp:forest-status($forest-id)//*:reindexing\n" +
+            ") = fn:true()";
+        StringHandle stringHandle;
+        String status;
+        int attempts = 1;
+        do{
+            sleep(1000L);
+            stringHandle = new StringHandle();
+            hubConfig.newStagingClient().newServerEval().xquery(query).eval().next().get(stringHandle);
```

Comment:
One tweak to make this simpler:

    String status = hubConfig...xquery(query).evalAs(String.class);

Then you don't need a StringHandle, nor do you need to fiddle with an iterator.